chainsync/chainsync.go

```diff
@@ -4,23 +4,18 @@ import (
 	"context"
 	"fmt"
 
-	"github.com/filecoin-project/lotus/chain/store"
 	"github.com/filecoin-project/lotus/chain/types"
+	"github.com/textileio/lotus-client/api/apistruct"
+	"github.com/textileio/lotus-client/chain/store"
 )
 
-// API provides an abstraction to a Filecoin full-node
-type API interface {
-	ChainGetPath(context.Context, types.TipSetKey, types.TipSetKey) ([]*store.HeadChange, error)
-	ChainGetGenesis(context.Context) (*types.TipSet, error)
-}
```

Comment:
Decided to drop these "abstractions" from modules. In the end they are always referring to concrete Lotus APIs.
Originally, I thought it would be useful to just narrow down _which_ APIs where used here.

Feels defining this API might be very much related in plans of supporting multiple clients or similar. It will take some works to build some reasonable-coherent generic API. But this is another story.